### DISCUSSION TITLE
Not supported by JOSE or your Javascript runtime.

### DISCUSSION THREAD

**Discussion by tannert44** [panva/jose#560] [Q&A] [answered]
alg RS256; is not supported either by JOSE or your javascript runtime

I see that this is an error in your documentation, but I have no idea why I am getting it or what to try to fix it. No one else seems to be having this issue. I've pretty much copied and pasted from your SignJWT documentation, so not sure what's going. The only difference is generated my own keys.

**Comment by panva** [comment]
There's nothing actionable in your message. No info about the runtime, no code, no way to reproduce...

**Reply by tannert44** [reply]
Fair enough. I did give a more detailed response once, decided to go back to another package, then came back to yours again so.  

Node.js v18.16.0 is the node version I have locally. Running the app using ts-node in development.  

How I am generating the keys:

`import * as jose from 'jose';
import * as fs from 'fs';

export default {};

const writeKeyPair = async () => {
  const { publicKey, privateKey } = await jose.generateKeyPair('RS256', { modulusLength: 2048, extractable: true });
  const privateKeyString = await jose.exportPKCS8(privateKey);
  const publicKeyString = await jose.exportSPKI(publicKey);
  fs.writeFileSync('./private.pem', privateKeyString);
  fs.writeFileSync('./public.pem', publicKeyString);
};

writeKeyPair();`


The function I am using to attempt to sign the JWT:

`const signJwt = async (sub: string, payload: jose.JWTPayload) => {
  //creating jwt token
  const secret = 'mysecret';
  const privateKey = await jose.importPKCS8(
    secret,
    'RS256'
  );

  const jwt =
  await new jose.SignJWT(payload)
    .setProtectedHeader({ typ: 'JWT', alg: 'RS256' })
    .setIssuedAt()
    .setSubject(sub)
    .setIssuer(process.env.COMMUNE_JWT_ISSUER)
    .setAudience(process.env.COMMUNE_JWT_AUDIENCE)
    .setExpirationTime(process.env.COMMUNE_JWT_EXPIRY)
    .sign(privateKey);

  return jwt;
};`

I get the error mentioned in the original post when it attempts to sign.

**Reply by panva** [reply]
Every supported nodejs version supports the RS256 algorithm, including v18.16.0. Your code does not exhibit the issue you're describing either...

**Reply by tannert44** [reply]
Well, I mean this is why I swapped packages. Like I said I pretty much copied and pasted from your documentation with the exception of generating the keys in a different manner. Could ts-node be the issue?  Just shooting in the dark.

Just to show you I am not telling you anything wrong, here's me catching the error. You'll just have to trust that it hits the catch block after running the sign method. I mean I wouldn't be getting one of your internal errors, if it wasn't actually coming from your packages code. 

![NodeVersion](https://github.com/panva/jose/assets/12992361/a0e0fe7a-29c5-453d-9c4c-25105e5fe7b3)
![ErrorAfterSign](https://github.com/panva/jose/assets/12992361/6ef58dd8-2b1d-4b53-b79f-8147d166a8a6)

**Reply by panva** [reply, marked as answer]
Why is there a semicolon after the algorithm name in the message?

**Reply by tannert44** [reply]
You're a life saver. They had a semicolon in the environment variable they had grabbing the algorithm.  I just assumed that was in your message.  I replaced it when I sent you the code, to avoid showing some of the names of me environment variables, and added it in correctly. Thanks a lot man.
